### PR TITLE
refactor(batch,agg): improve creation of agg states

### PR DESCRIPTION
**Pull request by richardchien**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Move the creation logic to `AggStateFactory::new` and simply clone the created initial agg state in `AggStateFactory::create_agg_state`. This can actually speed up `create_agg_state` and reduce memory consumption of `AggStateFactory` at the same time, for there is no need any more to walk through a huge amount of branches in `create_agg_state` and no need to store unnecessary values in `AggStateFactory`.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/common/src/array/utf8_array.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ use crate::array::ArrayBuilderImpl;
 use crate::buffer::{Bitmap, BitmapBuilder};
 
 /// `Utf8Array` is a collection of Rust Utf8 `String`s.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```
Comment (BugenZhao) [inline src/common/src/array/utf8_array.rs:29]: I'm not sure whether it's okay to derive `Clone` for all kinds of arrays. 🤔 cc @skyzh

File: src/expr/src/vector_op/agg/aggregator.rs [hunk 2]
```diff
@@ -90,103 +88,59 @@ impl AggStateFactory {
...
-                let input_type = DataType::from(arg.get_type()?);
-                let input_col_idx = arg.get_input()?.get_column_idx() as usize;
-                Ok(Self {
-                    agg_kind,
-                    return_type,
-                    input_type: Some(input_type),
+
+        let initial_agg_state = if agg_kind == AggKind::Count && prost.get_args().is_empty() {
+            Box::new(CountStar::new(return_type.clone(), filter))
+        } else if agg_kind == AggKind::StringAgg {
```
Comment (BugenZhao) [inline src/expr/src/vector_op/agg/aggregator.rs:94]: Will it be clearer if we write these lines with `match agg_kind`?

File: src/expr/src/vector_op/agg/aggregator.rs [hunk 3]
```diff
@@ -90,103 +88,59 @@ impl AggStateFactory {
...
+            let agg_arg = &prost.get_args()[0];
+            let delim_arg = &prost.get_args()[1];
+            debug_assert!(DataType::from(agg_arg.get_type().unwrap()) == DataType::Varchar);
```
Comment (BugenZhao) [inline src/expr/src/vector_op/agg/aggregator.rs:97]:
```suggestion
            assert_eq!(DataType::from(agg_arg.get_type().unwrap()), DataType::Varchar);
```

**Review by BugenZhao — COMMENTED** [review]
Generally LGTM

**Review by BugenZhao — APPROVED** [review]

**Review by liurenjie1024 — APPROVED** [review]

**Pull request merged**